UI & more

Jotschi-89 opened the pull request:
added Ui, added postfix names, fixed condition signal not working, ...

Jotschi89 committed:
added Ui, added postfix names, fixed condition signal not working, merged

RasmusStagsted commented on control.lua:
```diff
@@ -1,108 +1,103 @@
-    end
+script.on_event(defines.events.on_tick,
+  function(event)  
+    for trainStopUn, trainStopSetting in pairs(storage.dynamicTrainStopSettings) do
```
When loading an existing save, the mod crashes because "storage.dynamicTrainStopSettings" is not initialized.

Jotschi-89 replied:
The onInit code is only called on a new game or first time activating the mod. 
I think i might have newly invented the dynamicTrainStopSettings variable, so it will fail when loading a save that had the mod installed before. 
I guess it needs to be initalised at the on_tick methode too.

Jotschi-89 replied:
pushed a fix

RasmusStagsted commented:
I've just started a new game and build a train stop and a constant combinator (in editor mode), but I can't get the trains stop name to change.

Jotschi-89 commented:
> I've just started a new game and build a train stop and a constant combinator (in editor mode), but I can't get the trains stop name to change.

I just tested it, that does work for me. Did you see the checkboxes in the very right of the train station menu (they are unchecked by default)?

RasmusStagsted commented:
I like the way you've anchored the GUI element. It's definitely better than what I worked on, and I would like to add it as is. But I would still have a TODO, to add it to the existing GUI element.

Jotschi-89 commented:
As far as i understand it is not possible to change the exiting GUI. The options are to attach it somewhere or to completly rebuilt the existing UI which seems unreasonable. I am not completly satisfied with the position of it myself, so i am happy if you find a better version.

Jotschi89 committed:
added storage init in onLoad for save games with old mod version

RasmusStagsted commented:
> tested

I've checked both checkboxes and enabled the constant combinator (with a fast belt signal), but the name are not changing.

![Screenshot from 2024-12-05 21-40-25](https://github.com/user-attachments/assets/3ea1e82f-cafd-40bf-b5c3-91833b2f8ad4)

RasmusStagsted commented:
Maybe we should just focus on the GUI for now.
What you have made is much better than the GUI I came up with, so I'm very interested in merging that.
Would you mind making a PR of you GUI only? And maybe i could have you comply with my coding/formatting style (mainly snake_case and two spaces indents)? Then I would gladly approve your PR.

Jotschi-89 commented:
It seems to not work in map editor mode. 
It works as soon as i exit it with another /editor

Jotschi89 committed:
code styles, fixed problems with saves and old mod versions

Jotschi-89 commented:
The old mod version loading is a bit tricky. I changed the event for that again, as the onLoad was not possible because of desync issues. Now the check is run everytime the mod (or game) version changes. Also the renaming of the ui elements led to duplicated ui elements in existing saves, so i had to add an extra check for that. 

Changed the version number to 1.5, this is needed to make it work with old saves.

Jotschi89 committed:
added support for copy & past settings with shift click (not copy and past and blueprints)

Jotschi89 committed:
added blueprint and copy&paste support

RasmusStagsted commented:
I'm very happy with the work you have done.
I'm releasing a new patch on the mod portal tonight. I would like to add all the nice features and bug fixes you have made, but I also want a nice diff/commit history on github. Therefore I'm separating the features and bug fixes you have made, in separate commits, and pushing them to the master branch. Of cause you shall have all the credit you deserve. But to speed things up I will push it in my name (from my account), if you want, your name on the commits, we can later force push to master to edit the commit messages. And of cause I have added you to the auther list in the info.json file.
This is the commits I'm working on:  
- Remove debug code (Done).
- Fixed bug, when saves of old mods are loaded (Done).
- Use the entity id as key for the train stop table (Done).
- Fixed bug, when saves of old mods are loaded (Done).
- Fix bug, with enable/disable circuit condition (Done).
- Add support for post-fixed names (In progress).
- Add support for copy/paste (In progress).
- Add the GUI (In progress).

Jotschi-89 commented:
All fine for me. Better commits are for sure a good idea.
